platformminigbm.cpp

```diff
@@ -83,20 +88,42 @@ int DrmMinigbmImporter::ImportBuffer(buffer_handle_t handle, hwc_drm_bo_t *bo) {
   bo->width = gr_handle->width;
   bo->height = gr_handle->height;
   bo->hal_format = gr_handle->droid_format;
-  bo->format = gr_handle->format;
+  if (need_modifier)
+    bo->format = DRM_FORMAT_NV12;
+  else
+    bo->format = gr_handle->format;
   bo->usage = gr_handle->usage;
   bo->pixel_stride = gr_handle->pixel_stride;
-  bo->pitches[0] = gr_handle->strides[0];
-  bo->offsets[0] = gr_handle->offsets[0];
-  bo->gem_handles[0] = gem_handle;
 
-  ret = drmModeAddFB2(drm_->fd(), bo->width, bo->height, bo->format,
-                      bo->gem_handles, bo->pitches, bo->offsets, &bo->fb_id, 0);
-  if (ret) {
-    ALOGE("could not create drm fb %d", ret);
-    return ret;
+  uint32_t numPlanes = gr_handle->base.numFds;
+  for (uint32_t i = 0; i < numPlanes; ++i) {
+    bo->pitches[i] = gr_handle->strides[i];
+    bo->offsets[i] = gr_handle->offsets[i];
+    bo->gem_handles[i] = gem_handle;
   }
 
+  if (need_modifier == false) {
+    ret = drmModeAddFB2(drm_->fd(), bo->width, bo->height, bo->format,
+                        bo->gem_handles, bo->pitches, bo->offsets, &bo->fb_id, 0);
+  }
+  else {
+    uint32_t flag = DRM_MODE_FB_MODIFIERS;
+    uint64_t modifiers[HWC_DRM_BO_MAX_PLANES];
+    uint32_t numPlanes = gr_handle->base.numFds;
+
+    for (uint32_t i = 0; i < numPlanes; i++)
+      modifiers[i] = I915_FORMAT_MOD_Y_TILED;
```

Comment:
need add format check, if format is DRM_FORMAT_NV12_Y_TILED_INTEL, then modifiers set to I915_FORMAT_MOD_Y_TILED.
And we only support DRM_FORMAT_NV12_Y_TILED_INTEL, can add the assert

Reply:
Done.